File: Geek Assistant (Android AIO)/Modules/General/Action/GA_Reset.vb
Imports System.IO

Module GA_Reset

    Public Sub Run(Data As Boolean, logs As Boolean)
        ErrorCodeTrack("")
        LogEvent("Reset Geek Assistant", 2)
        If Not GA_infoAsk.Run("Reset Geek Assistant",
                          $"Are you sure you want to {GA_Settings.willClear}?",
                          "Continue", "Cancel") Then
            LogAppendText("Reset Geek Assistant Cancelled.", 1)
            Exit Sub
        End If

        LogAppendText($"Resetting Geek Assistant... {vbCrLf}({GA_Settings.willClear})", 1)
        Try
            Dim notify As String = "Process Complete. "
            If Data Then
                ErrorCodeTrack("GAr-S") 'GA reset - Settings
                My.Settings.Reset()
                My.Settings.Save()
                notify &= $"{vbCrLf}Do you want to relaunch Geek Assistant?{vbCrLf}{vbCrLf}{vbCrLf}" &
                           "⚠ Warning: Relaunching will recreate some files needed to run Geek Assistant."
            End If

            If logs Then
                ErrorCodeTrack("GAr-L") 'GA reset - Logs
                If Directory.Exists(GA_logs) Then
                    For Each foundName As String In My.Computer.FileSystem.GetFiles(GA_logs, FileIO.SearchOption.SearchAllSubDirectories, "*.*")
                        If File.Exists($"{foundName}\*.*") Then
                            My.Computer.FileSystem.DeleteFile(foundName, FileIO.UIOption.OnlyErrorDialogs, FileIO.RecycleOption.DeletePermanently)
                        End If
                    Next
                    My.Computer.FileSystem.DeleteDirectory(GA_logs, FileIO.UIOption.OnlyErrorDialogs, FileIO.RecycleOption.DeletePermanently)
                End If
            End If

            LogAppendText(notify, 1)
            If Data Then
                My.Settings.VerboseLoggingPrompt = False 'disable to avoid asking on exit
                If GA_infoAsk.Run("Reset Geek Assistant", notify, "Relaunch", "Exit") Then
                    My.Settings.VerboseLoggingPrompt = True ' enable again (true is default)
                    Application.Restart()
                Else
                    My.Settings.VerboseLoggingPrompt = True ' enable again (true is default)
                    Application.Exit()
                End If
            Else
                MessageBox.Show(notify, "Reset Geek Assistant", MessageBoxButtons.OK, MessageBoxIcon.Information)
            End If


        Catch e As Exception
            DoMsg(10, "Error while resetting.", 3, e.ToString)
        End Try
    End Sub

End Module

File: Geek Assistant (Android AIO)/Modules/General/Action/GA_Reset.cs
using GeekAssistant.Forms;
using System;
using System.IO;
using System.Windows.Forms;

internal static partial class GA_Reset {
    public static void Run(bool Data, bool logs) {
        common.ErrorInfo.code = "GAr-00";
        GA_Log.LogEvent("Reset Geek Assistant", 2);
        if (!GA_infoAsk.Run("Reset Geek Assistant", 
                              $"Are you sure you want to {Settings.willClear}?", 
                            "Continue", "Cancel")) {
            GA_Log.LogAppendText("Reset Geek Assistant Cancelled.", 1);
            return;
        }

        GA_Log.LogAppendText($"Resetting Geek Assistant... \n({Settings.willClear})", 1);
        try {
            string notify = "Process Complete. ";
            if (Data) {
                common.ErrorInfo.code = "GAr-S"; // GA reset - Settings
                common.S.Reset();
                common.S.Save();
                notify += $"\nDo you want to relaunch Geek Assistant?\n\n\n" + "⚠ Warning: Relaunching will recreate some files needed to run Geek Assistant.";
            }

            if (logs) {
                common.ErrorInfo.code = "GAr-L"; // GA reset - Logs
                if (Directory.Exists(common.GA_logs)) {
                    foreach (string foundName in Microsoft.VisualBasic.FileIO.FileSystem.GetFiles(common.GA_logs, Microsoft.VisualBasic.FileIO.SearchOption.SearchAllSubDirectories, "*.*")) {
                        if (File.Exists($@"{foundName}\*.*")) {
                            Microsoft.VisualBasic.FileIO.FileSystem.DeleteFile(foundName, Microsoft.VisualBasic.FileIO.UIOption.OnlyErrorDialogs, Microsoft.VisualBasic.FileIO.RecycleOption.DeletePermanently);
                        }
                    }

                    Microsoft.VisualBasic.FileIO.FileSystem.DeleteDirectory(common.GA_logs, Microsoft.VisualBasic.FileIO.UIOption.OnlyErrorDialogs, Microsoft.VisualBasic.FileIO.RecycleOption.DeletePermanently);
                }
            }

            GA_Log.LogAppendText(notify, 1);
            if (Data) {
                common.S.VerboseLoggingPrompt = false; // disable to avoid asking on exit
                if (GA_infoAsk.Run("Reset Geek Assistant", notify, "Relaunch", "Exit")) {
                    common.S.VerboseLoggingPrompt = true; // enable again (true is default)
                    Application.Restart();
                } else {
                    common.S.VerboseLoggingPrompt = true; // enable again (true is default)
                    Application.Exit();
                }
            } else MessageBox.Show(notify, "Reset Geek Assistant", MessageBoxButtons.OK, MessageBoxIcon.Information);

        } catch (Exception e) { GA_Msg.DoMsg(10, "Error while resetting.", 3, e.ToString()); }
         
    }
}